### PR TITLE
[Modules] CDN Profile afd resources

### LINKED ISSUE #4042: [Feature Request]: CDN Profile multiple endpoints with multiple origins and routes
### Description

Currently, the CDN profile module does not have support for the flowing resource creation: 

- Multiple Endpoints with Origin Groups and routes
- Custom domains
- Rules Sets
- Secrets

I think the CDN profile module is the [classic](https://learn.microsoft.com/en-us/azure/cdn/create-profile-endpoint-bicep?toc=%2Fazure%2Ffrontdoor%2FTOC.json&tabs=CLI) one. 

We want to use the [Front Door Standard/Premium](https://learn.microsoft.com/en-us/azure/frontdoor/create-front-door-bicep?tabs=CLI). Can you please add support?

### PR DESCRIPTION
**Pull request by ppotturi**
# Description
Added support to create Standard/Premium front door with multiple afd endpoints, routes, custom domains, secrets and rule sets. 

Fixes  #4042 

### Pipeline references

| Pipeline |
| - |
|[CDN - Profiles](https://github.com/defra-adp-sandpit/ResourceModules/actions/workflows/ms.cdn.profiles.yml) |

## Type of Change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

File: modules/cdn/profile/.test/afd/dependencies.bicep [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  location: location
+}
+
+@description('The resource ID of the created Storage Account.')
+output storageAccountResourceId string = storageAccount.id
+
+@description('The name of the created Storage Account.')
+output storageAccountName string = storageAccount.name
+
+@description('The resource ID of the created Managed Identity.')
```
Comment (AlexanderSehr) [inline modules/cdn/profile/.test/afd/dependencies.bicep:37]:
```suggestion
@description('The principal ID of the created Managed Identity.')
```

File: modules/cdn/profile/afdEndpoint/main.bicep [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    customDomainName: contains(route, 'customDomainName') ? route.customDomainName : ''
+    enabledState: contains(route, 'enabledState') ? route.enabledState : 'Enabled'
+    forwardingProtocol: contains(route, 'forwardingProtocol') ? route.forwardingProtocol : 'MatchRequest'
+    httpsRedirect: contains(route, 'httpsRedirect') ? route.httpsRedirect : 'Enabled'
+    linkToDefaultDomain: contains(route, 'linkToDefaultDomain') ? route.linkToDefaultDomain : 'Enabled'
+    originGroupName: contains(route, 'originGroupName') ? route.originGroupName : ''
+    originPath: contains(route, 'originPath') ? route.originPath : ''
+    patternsToMatch: contains(route, 'patternsToMatch') ? route.patternsToMatch : []
+    ruleSets: contains(route, 'ruleSets') ? route.ruleSets : []
+    supportedProtocols: contains(route, 'supportedProtocols') ? route.supportedProtocols : []
```
Comment (AlexanderSehr) [inline modules/cdn/profile/afdEndpoint/main.bicep:82]:
```suggestion
    supportedProtocols: contains(route, 'supportedProtocols') ? route.supportedProtocols : []
    enableDefaultTelemetry: enableReferencedModulesTelemetry
```

File: modules/cdn/profile/afdEndpoint/main.bicep [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  'Enabled'
+  'Disabled'
+])
+param enabledState string = 'Enabled'
+
+@description('Optional. The list of routes for this AFD Endpoint.')
+param routes array = []
+
+@description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
+param enableDefaultTelemetry bool = true
```
Comment (AlexanderSehr) [inline modules/cdn/profile/afdEndpoint/main.bicep:37]:
```suggestion
param enableDefaultTelemetry bool = true

var enableReferencedModulesTelemetry = false
```

File: modules/cdn/profile/afdEndpoint/route/main.bicep [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+}
+
+resource profile 'Microsoft.Cdn/profiles@2023-05-01' existing = {
+  name: profileName
+}
+
+resource afd_endpoint 'Microsoft.Cdn/profiles/afdEndpoints@2023-05-01' existing = {
+  name: afdEndpointName
+  parent: profile
+}
```
Comment (AlexanderSehr) [inline modules/cdn/profile/afdEndpoint/route/main.bicep:87]:
```suggestion
resource profile 'Microsoft.Cdn/profiles@2023-05-01' existing = {
  name: profileName
  
  resource afd_endpoint 'afdEndpoints@2023-05-01' existing = {
    name: afdEndpointName
  }
}
```

File: modules/cdn/profile/afdEndpoint/route/main.bicep [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+
+resource rule_set 'Microsoft.Cdn/profiles/ruleSets@2023-05-01' existing = [for ruleSet in ruleSets: {
+  name: ruleSet.name
+  parent: profile
+}]
+
+resource afd_endpoint_route 'Microsoft.Cdn/profiles/afdEndpoints/routes@2023-05-01' = {
+  name: name
+  parent: afd_endpoint
```
Comment (AlexanderSehr) [inline modules/cdn/profile/afdEndpoint/route/main.bicep:106]:
```suggestion
  parent: profile::afd_endpoint
```

File: modules/cdn/profile/afdEndpoint/route/main.bicep [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+
+resource originGroup 'Microsoft.Cdn/profiles/originGroups@2023-05-01' existing = {
+  name: originGroupName
+  parent: profile
+}
+
+resource rule_set 'Microsoft.Cdn/profiles/ruleSets@2023-05-01' existing = [for ruleSet in ruleSets: {
+  name: ruleSet.name
+  parent: profile
+}]
```
Comment (AlexanderSehr) [inline modules/cdn/profile/afdEndpoint/route/main.bicep:102]:
```suggestion
resource profile 'Microsoft.Cdn/profiles@2023-05-01' existing = {
  name: profileName
  
  resource afd_endpoint 'afdEndpoints@2023-05-01' existing = {
    name: afdEndpointName
  }
  
  resource custom_domain 'customDomains@2023-05-01' existing = if (!empty(customDomainName)) {
    name: customDomainName
  }

  resource originGroup 'originGroups@2023-05-01' existing = {
    name: originGroupName
  }
  
  resource rule_set 'ruleSets@2023-05-01' existing = [for ruleSet in ruleSets: {
    name: ruleSet.name
  }]
}
```

File: modules/cdn/profile/afdEndpoint/route/main.bicep [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+      id: originGroup.id
+    }
+    originPath: !empty(originPath) ? originPath : null
+    patternsToMatch: patternsToMatch
+    ruleSets: [for (item, index) in ruleSets: {
+      id: rule_set[index].id
+    }]
+    supportedProtocols: !empty(supportedProtocols) ? supportedProtocols : null
+  }
+}
```
Comment (AlexanderSehr) [inline modules/cdn/profile/afdEndpoint/route/main.bicep:126]:
```suggestion
resource afd_endpoint_route 'Microsoft.Cdn/profiles/afdEndpoints/routes@2023-05-01' = {
  name: name
  parent: profile::afd_endpoint
  properties: {
    cacheConfiguration: !empty(cacheConfiguration) ? cacheConfiguration : null
    customDomains: !empty(customDomainName) ? [ {
        id: profile::custom_domain.id
      } ] : []
    enabledState: enabledState
    forwardingProtocol: forwardingProtocol
    httpsRedirect: httpsRedirect
    linkToDefaultDomain: linkToDefaultDomain
    originGroup: {
      id: profile::originGroup.id
    }
    originPath: !empty(originPath) ? originPath : null
    patternsToMatch: patternsToMatch
    ruleSets: [for (item, index) in ruleSets: {
      id: profile::rule_set[index].id
    }]
    supportedProtocols: !empty(supportedProtocols) ? supportedProtocols : null
  }
}
```

File: modules/cdn/profile/customdomain/main.bicep [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+}
+
+resource profile 'Microsoft.Cdn/profiles@2023-05-01' existing = {
+  name: profileName
+}
+
+resource profile_secrect 'Microsoft.Cdn/profiles/secrets@2023-05-01' existing = if (!empty(secretName)) {
+  name: secretName
+  parent: profile
+}
```
Comment (AlexanderSehr) [inline modules/cdn/profile/customdomain/main.bicep:62]:
```suggestion
resource profile 'Microsoft.Cdn/profiles@2023-05-01' existing = {
  name: profileName

  resource profile_secrect 'secrets@2023-05-01' existing = if (!empty(secretName)) {
    name: secretName
  }
}
```

File: modules/cdn/profile/customdomain/main.bicep [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+    extendedProperties: !empty(extendedProperties) ? extendedProperties : null
+    hostName: hostName
+    preValidatedCustomDomainResourceId: !empty(preValidatedCustomDomainResourceId) ? {
+      id: preValidatedCustomDomainResourceId
+    } : null
+    tlsSettings: {
+      certificateType: certificateType
+      minimumTlsVersion: minimumTlsVersion
+      secret: !(empty(secretName)) ? {
+        id: profile_secrect.id
```
Comment (AlexanderSehr) [inline modules/cdn/profile/customdomain/main.bicep:80]:
```suggestion
        id: profile::profile_secrect.id
```

File: modules/cdn/profile/customdomain/main.bicep [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+resource profile 'Microsoft.Cdn/profiles@2023-05-01' existing = {
+  name: profileName
+}
+
+resource profile_secrect 'Microsoft.Cdn/profiles/secrets@2023-05-01' existing = if (!empty(secretName)) {
+  name: secretName
+  parent: profile
+}
+
+resource prifile_custom_domain 'Microsoft.Cdn/profiles/customDomains@2023-05-01' = {
```
Comment (AlexanderSehr) [inline modules/cdn/profile/customdomain/main.bicep:64]:
```suggestion
resource profile_custom_domain 'Microsoft.Cdn/profiles/customDomains@2023-05-01' = {
```

File: modules/cdn/profile/customdomain/main.bicep [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+      } : null
+    }
+  }
+}
+
+@description('The name of the custom domain.')
+output name string = prifile_custom_domain.name
+
+@description('The resource id of the custom domain.')
+output resourceId string = prifile_custom_domain.id
```
Comment (AlexanderSehr) [inline modules/cdn/profile/customdomain/main.bicep:90]:
```suggestion
output name string = profile_custom_domain.name

@description('The resource id of the custom domain.')
output resourceId string = profile_custom_domain.id
```

File: modules/cdn/profile/main.bicep [hunk 12]
```diff
@@ -9,23 +9,19 @@ param name string
...
+  'Standard_AvgBandWidth_ChinaCdn'
+  'Standard_AzureFrontDoor'
   'Standard_ChinaCdn'
   'Standard_Microsoft'
-  'Premium_Verizon'
-  'Premium_Akamai'
-  'Premium_ChinaCdn'
-  'Premium_Microsoft'
-  'Custom_Verizon'
-  'Custom_Akamai'
```
Comment (AlexanderSehr) [inline modules/cdn/profile/main.bicep:21]: I see some were removed. Are they 'not supported'?

File: modules/cdn/profile/main.bicep [hunk 13]
```diff
@@ -115,6 +126,82 @@ module profile_endpoint 'endpoint/main.bicep' = if (!empty(endpointProperties))
   }
 }
 
+module profile_secret 'secret/main.bicep' = [for secret in secrets: {
+  name: '${uniqueString(deployment().name, secret.name)}-Profile-Secret'
```
Comment (AlexanderSehr) [inline modules/cdn/profile/main.bicep:130]:
```suggestion
module profile_secret 'secret/main.bicep' = [for (secret, index) in secrets: {
  name: '${uniqueString(deployment().name, secret.name)}-Profile-Secret-${index}'
```

File: modules/cdn/profile/main.bicep [hunk 14]
```diff
@@ -115,6 +126,82 @@ module profile_endpoint 'endpoint/main.bicep' = if (!empty(endpointProperties))
...
+    secretSourceResourceId: secret.secretSourceResourceId
+    subjectAlternativeNames: contains(secret, 'subjectAlternativeNames') ? secret.subjectAlternativeNames : []
+    useLatestVersion: contains(secret, 'useLatestVersion') ? secret.useLatestVersion : false
+    secretVersion: secret.secretVersion
+    enableDefaultTelemetry: enableReferencedModulesTelemetry
+  }
+}]
+
+module profile_custom_domain 'customdomain/main.bicep' = [for customDomain in customDomains: {
+  name: '${uniqueString(deployment().name, customDomain.name)}-CustomDomain'
```
Comment (AlexanderSehr) [inline modules/cdn/profile/main.bicep:144]:
```suggestion
module profile_custom_domain 'customdomain/main.bicep' = [for (customDomain, index) in customDomains: {
  name: '${uniqueString(deployment().name)}-CustomDomain-${index}'
```

File: modules/cdn/profile/main.bicep [hunk 15]
```diff
@@ -115,6 +126,82 @@ module profile_endpoint 'endpoint/main.bicep' = if (!empty(endpointProperties))
...
+    certificateType: customDomain.certificateType
+    minimumTlsVersion: contains(customDomain, 'minimumTlsVersion') ? customDomain.minimumTlsVersion : 'TLS12'
+    preValidatedCustomDomainResourceId: contains(customDomain, 'preValidatedCustomDomainResourceId') ? customDomain.preValidatedCustomDomainResourceId : ''
+    secretName: contains(customDomain, 'secretName') ? customDomain.secretName : ''
+    enableDefaultTelemetry: enableReferencedModulesTelemetry
+  }
+}]
+
+module profile_origionGroup 'origingroup/main.bicep' = [for origingroup in origionGroups: {
+  name: '${uniqueString(deployment().name, origingroup.name)}-Profile-OrigionGroup'
```
Comment (AlexanderSehr) [inline modules/cdn/profile/main.bicep:163]:
```suggestion
module profile_origionGroup 'origingroup/main.bicep' = [for (origingroup, index) in origionGroups: {
  name: '${uniqueString(deployment().name)}-Profile-OrigionGroup-${index}'
```

File: modules/cdn/profile/main.bicep [hunk 16]
```diff
@@ -115,6 +126,82 @@ module profile_endpoint 'endpoint/main.bicep' = if (!empty(endpointProperties))
...
+    loadBalancingSettings: origingroup.loadBalancingSettings
+    sessionAffinityState: contains(origingroup, 'sessionAffinityState') ? origingroup.sessionAffinityState : 'Disabled'
+    trafficRestorationTimeToHealedOrNewEndpointsInMinutes: contains(origingroup, 'trafficRestorationTimeToHealedOrNewEndpointsInMinutes') ? origingroup.trafficRestorationTimeToHealedOrNewEndpointsInMinutes : 10
+    origins: origingroup.origins
+    enableDefaultTelemetry: enableReferencedModulesTelemetry
+  }
+}]
+
+module profile_ruleSet 'ruleset/main.bicep' = [for ruleSet in ruleSets: {
+  name: '${uniqueString(deployment().name, ruleSet.name)}-Profile-RuleSet'
```
Comment (AlexanderSehr) [inline modules/cdn/profile/main.bicep:177]:
```suggestion
module profile_ruleSet 'ruleset/main.bicep' = [for (ruleSet, index) in ruleSets: {
  name: '${uniqueString(deployment().name)}-Profile-RuleSet-${index}'
```

File: modules/cdn/profile/main.bicep [hunk 17]
```diff
@@ -115,6 +126,82 @@ module profile_endpoint 'endpoint/main.bicep' = if (!empty(endpointProperties))
...
+  params: {
+    name: ruleSet.name
+    profileName: profile.name
+    rules: ruleSet.rules
+    enableDefaultTelemetry: enableReferencedModulesTelemetry
+  }
+}]
+
+module profile_afdEndpoint 'afdEndpoint/main.bicep' = [for afdEndpoint in afdEndpoints: {
+  name: '${uniqueString(deployment().name, afdEndpoint.name)}-Profile-AfdEndpoint'
```
Comment (AlexanderSehr) [inline modules/cdn/profile/main.bicep:187]:
```suggestion
module profile_afdEndpoint 'afdEndpoint/main.bicep' = [for (afdEndpoint, index) in afdEndpoints: {
  name: '${uniqueString(deployment().name)}-Profile-AfdEndpoint-${index}'
```

File: modules/cdn/profile/origingroup/main.bicep [hunk 18]
```diff
@@ -0,0 +1,87 @@
...
+  properties: {
+    healthProbeSettings: !empty(healthProbeSettings) ? healthProbeSettings : null
+    loadBalancingSettings: loadBalancingSettings
+    sessionAffinityState: sessionAffinityState
+    trafficRestorationTimeToHealedOrNewEndpointsInMinutes: trafficRestorationTimeToHealedOrNewEndpointsInMinutes
+  }
+}
+
+module origin 'origin/main.bicep' = [for origion in origins: {
+  name: '${uniqueString(deployment().name, origion.name)}-OriginGroup-Origin-${origion.name}'
```
Comment (AlexanderSehr) [inline modules/cdn/profile/origingroup/main.bicep:61]:
```suggestion
module origin 'origin/main.bicep' = [for (origion, index) in origins: {
  name: '${uniqueString(deployment().name)}-OriginGroup-Origin-${origion.name}-${index}'
```

File: modules/cdn/profile/origingroup/main.bicep [hunk 19]
```diff
@@ -0,0 +1,87 @@
...
+  }
+}]
+
+@description('The name of the origin group.')
+output name string = originGroup.name
+
+@description('The resource id of the origin group.')
+output resourceId string = originGroup.id
+@description('The name of the resource group the origin group was created in.')
+output resourceGroupName string = resourceGroup().name
```
Comment (AlexanderSehr) [inline modules/cdn/profile/origingroup/main.bicep:84]:
```suggestion
@description('The resource id of the origin group.')
output resourceId string = originGroup.id

@description('The name of the resource group the origin group was created in.')
output resourceGroupName string = resourceGroup().name
```

File: modules/cdn/profile/origingroup/main.bicep [hunk 20]
```diff
@@ -0,0 +1,87 @@
...
+
+@description('The location the resource was deployed into.')
+output location string = profile.location
```
Comment (AlexanderSehr) [inline modules/cdn/profile/origingroup/main.bicep:87]:
```suggestion
output location string = originGroup.location
```

Comment (AlexanderSehr) [reply modules/cdn/profile/origingroup/main.bicep:87]: If supported by resource

File: modules/cdn/profile/origingroup/origin/main.bicep [hunk 21]
```diff
@@ -0,0 +1,92 @@
...
+}
+
+resource profile 'Microsoft.Cdn/profiles@2023-05-01' existing = {
+  name: profileName
+}
+
+resource originGroup 'Microsoft.Cdn/profiles/originGroups@2023-05-01' existing = {
+  name: originGroupName
+  parent: profile
+}
```
Comment (AlexanderSehr) [inline modules/cdn/profile/origingroup/origin/main.bicep:67]:
```suggestion
resource profile 'Microsoft.Cdn/profiles@2023-05-01' existing = {
  name: profileName

  resource originGroup 'originGroups@2023-05-01' existing = {
    name: originGroupName
  }
}
```

File: modules/cdn/profile/origingroup/origin/main.bicep [hunk 22]
```diff
@@ -0,0 +1,92 @@
...
+
+resource origin 'Microsoft.Cdn/profiles/originGroups/origins@2023-05-01' = {
+  name: name
+  parent: originGroup
```
Comment (AlexanderSehr) [inline modules/cdn/profile/origingroup/origin/main.bicep:71]:
```suggestion
  parent: profile::originGroup
```

File: modules/cdn/profile/origingroup/origin/main.bicep [hunk 23]
```diff
@@ -0,0 +1,92 @@
...
+    priority: priority
+    sharedPrivateLinkResource: !empty(sharedPrivateLinkResource) ? sharedPrivateLinkResource : null
+    weight: weight
+  }
+}
+
+@description('The name of the origin.')
+output name string = origin.name
+
+@description('The id of the origin.')
```
Comment (AlexanderSehr) [inline modules/cdn/profile/origingroup/origin/main.bicep:88]:
```suggestion
@description('The resource id of the origin.')
```

File: modules/cdn/profile/ruleset/main.bicep [hunk 24]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+resource profile 'Microsoft.Cdn/profiles@2023-05-01' existing = {
+  name: profileName
+}
+
+resource rule_set 'Microsoft.Cdn/profiles/ruleSets@2023-05-01' = {
+  name: name
+  parent: profile
+}
```
Comment (AlexanderSehr) [inline modules/cdn/profile/ruleset/main.bicep:36]:
```suggestion
resource profile 'Microsoft.Cdn/profiles@2023-05-01' existing = {
  name: profileName

  resource rule_set 'ruleSets@2023-05-01' = {
    name: name
  }
}
```

File: modules/cdn/profile/ruleset/main.bicep [hunk 25]
```diff
@@ -0,0 +1,58 @@
...
+
+module rule 'rule/main.bicep' = [for rule in rules: {
+  name: rule.name
```
Comment (AlexanderSehr) [inline modules/cdn/profile/ruleset/main.bicep:39]:
```suggestion
module rule 'rule/main.bicep' = [for (rule, index) in rules: {
  name: '${uniqueString(deployment().name, location)}-RuleSet-Rule-${index}'
```

File: modules/cdn/profile/ruleset/rule/main.bicep [hunk 26]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+resource profile 'Microsoft.Cdn/profiles@2023-05-01' existing = {
+  name: profileName
+}
+
+resource rule_set 'Microsoft.Cdn/profiles/ruleSets@2023-05-01' existing = {
+  name: ruleSetName
+  parent: profile
+}
```
Comment (AlexanderSehr) [inline modules/cdn/profile/ruleset/rule/main.bicep:52]:
```suggestion
resource profile 'Microsoft.Cdn/profiles@2023-05-01' existing = {
  name: profileName
  
  resource rule_set 'ruleSets@2023-05-01' existing = {
    name: ruleSetName
  }
}
```

File: modules/cdn/profile/ruleset/rule/main.bicep [hunk 27]
```diff
@@ -0,0 +1,72 @@
...
+
+resource rule_set_rule 'Microsoft.Cdn/profiles/ruleSets/rules@2023-05-01' = {
+  name: name
+  parent: rule_set
```
Comment (AlexanderSehr) [inline modules/cdn/profile/ruleset/rule/main.bicep:56]:
```suggestion
  parent: profile::rule_set
```

File: modules/cdn/profile/ruleset/main.bicep [hunk 28]
```diff
@@ -0,0 +1,58 @@
...
+  params: {
+    profileName: profileName
+    ruleSetName: name
+    name: rule.name
+    order: rule.order
+    actions: rule.actions
+    conditions: contains(rule, 'conditions') ? rule.conditions : []
+    matchProcessingBehavior: contains(rule, 'matchProcessingBehavior') ? rule.matchProcessingBehavior : 'Continue'
```
Comment (AlexanderSehr) [inline modules/cdn/profile/ruleset/main.bicep:47]:
```suggestion
    matchProcessingBehavior: contains(rule, 'matchProcessingBehavior') ? rule.matchProcessingBehavior : 'Continue'
    enableDefaultTelemetry: enableReferencedModulesTelemetry
```

File: modules/cdn/profile/ruleset/main.bicep [hunk 29]
```diff
@@ -0,0 +1,58 @@
...
+param name string
+
+@description('Required. The name of the CDN profile.')
+param profileName string
+
+@description('Optinal. The rules to apply to the rule set.')
+param rules array = []
+
+@description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
+param enableDefaultTelemetry bool = true
```
Comment (AlexanderSehr) [inline modules/cdn/profile/ruleset/main.bicep:15]:
```suggestion
param enableDefaultTelemetry bool = true
```
```suggestion
param enableDefaultTelemetry bool = true

var enableReferencedModulesTelemetry = false
```

File: modules/cdn/profile/origingroup/main.bicep [hunk 30]
```diff
@@ -0,0 +1,87 @@
...
+param sessionAffinityState string = 'Disabled'
+
+@description('Optional. Time in minutes to shift the traffic to the endpoint gradually when an unhealthy endpoint comes healthy or a new endpoint is added. Default is 10 mins.')
+param trafficRestorationTimeToHealedOrNewEndpointsInMinutes int = 10
+
+@description('Required. The list of origins within the origin group.')
+param origins array = []
+
+@description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
+param enableDefaultTelemetry bool = true
```
Comment (AlexanderSehr) [inline modules/cdn/profile/origingroup/main.bicep:31]:
```suggestion
param enableDefaultTelemetry bool = true
```
```suggestion
param enableDefaultTelemetry bool = true

var enableReferencedModulesTelemetry = false
```

File: modules/cdn/profile/origingroup/main.bicep [hunk 31]
```diff
@@ -0,0 +1,87 @@
...
+    hostName: origion.hostName
+    originGroupName: originGroup.name
+    enabledState: contains(origion, 'enabledState') ? origion.enabledState : 'Enabled'
+    enforceCertificateNameCheck: contains(origion, 'enforceCertificateNameCheck') ? origion.enforceCertificateNameCheck : true
+    httpPort: contains(origion, 'httpPort') ? origion.httpPort : 80
+    httpsPort: contains(origion, 'httpsPort') ? origion.httpsPort : 443
+    originHostHeader: contains(origion, 'originHostHeader') ? origion.originHostHeader : origion.hostName
+    priority: contains(origion, 'priority') ? origion.priority : 1
+    weight: contains(origion, 'weight') ? origion.weight : 1000
+    sharedPrivateLinkResource: contains(origion, 'sharedPrivateLinkResource') ? origion.sharedPrivateLinkResource : null
```
Comment (AlexanderSehr) [inline modules/cdn/profile/origingroup/main.bicep:74]:
```suggestion
    sharedPrivateLinkResource: contains(origion, 'sharedPrivateLinkResource') ? origion.sharedPrivateLinkResource : null
    enableDefaultTelemetry: enableReferencedModulesTelemetry
```

File: modules/cdn/profile/afdEndpoint/route/main.bicep [hunk 32]
```diff
@@ -0,0 +1,135 @@
...
+
+@description('Optional. The rule sets of the rule. The rule sets must be defined in the profile ruleSets.')
+param ruleSets array = []
+
+@allowed([ 'Http', 'Https' ])
+@description('Optional. The supported protocols of the rule.')
+param supportedProtocols array = []
+
+@description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
+param enableDefaultTelemetry bool = true
```
Comment (AlexanderSehr) [inline modules/cdn/profile/afdEndpoint/route/main.bicep:66]:
```suggestion
param enableDefaultTelemetry bool = true
```
```suggestion
param enableDefaultTelemetry bool = true

var enableReferencedModulesTelemetry = false
```

File: modules/cdn/profile/main.bicep (hunk 12 above)
Comment (ppotturi) [reply modules/cdn/profile/main.bicep:21]: They are not suported. This is the supoprted list.

File: modules/cdn/profile/origingroup/main.bicep (hunk 20 above)
Comment (ppotturi) [reply modules/cdn/profile/origingroup/main.bicep]: Location is not supported by Origin Group resource

**Comment by ppotturi** [comment]
@AlexanderSehr Resolved all your comments. Thank you.

**Comment by AlexanderSehr** [comment]
> @AlexanderSehr Resolved all your comments. Thank you.

Thanks @ppotturi and thanks for the great contribution. Thorough work. Can you do me the have to run the pipeline again? Using the "suggestions" feature I can never be sure I updated all references properly

**Comment by ppotturi** [comment]
> > @AlexanderSehr Resolved all your comments. Thank you.
> 
> Thanks @ppotturi and thanks for the great contribution. Thorough work. Can you do me the have to run the pipeline again? Using the "suggestions" feature I can never be sure I updated all references properly

I  ran the pipeline with the latest changes. All good. you can see the latest run in the above pipeline link.

**Review by AlexanderSehr — APPROVED** [review]
Much appreciated 💪 thank your for your great contribution

**Pull request merged**